mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,449 @@
+            return artifact
+
+    @staticmethod
+    def _log_run(
+        context: MLClientCtx, run: mlflow.entities.Run, is_offline: bool = False
+    ):
+        """
+        Log the given MLFlow run to MLRun.
+
+        :param context:    Current MLRun context or project.
+        :param run: MLFlow Run to log. Can be given as a `Run` object or as a run id.
+        :param is_offline: True if logging an offline run (importing), False if online run (tracking)
+        """
+        client = mlflow.MlflowClient()
+
+        # Create a temporary directory to log all data temporarily:
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            # When running in offline mode (manually importing) we want to make sure there is an `artifact_path` set in
+            # the context, if not we will set it to this temporary directory so the user won't have garbage in his
+            # local work dir:
+            if is_offline and not context.artifact_path:
+                context.artifact_path = tmp_dir
+
+            # Set new MLRun related tags to the MLFlow run:
+            client.set_tag(
+                run_id=run.info.run_id, key="mlrun-context-id", value=context.uid
+            )
+            client.set_tag(
+                run_id=run.info.run_id, key="mlrun-project-name", value=context.project
+            )
+
+            # Get the MLFlow run's tags and save them as labels:
+            context.set_label(key="mlflow-user", value=run.data.tags.get("mlflow.user"))
+            context.set_label(
+                key="mlflow-run-name", value=run.data.tags.get("mlflow.runName")
+            )
+            context.set_label(key="mlflow-run-id", value=run.info.run_id)
+            context.set_label(key="mlflow-experiment-id", value=run.info.experiment_id)
+
+            # Get the MLFlow run's parameters and save as job parameters:
+            context._parameters.update(run.data.params)
+
+            # Get the MLFlow run's metrics and log them as results:
+            results = run.data.metrics
+            context.log_results(results=results)
+
+            # Import the MLFlow run's artifacts to MLRun (model are logged after the rest of artifacts
+            # so the artifacts can be registered as extra data in the models):
+            artifacts = {}
+            model_paths = []
+            for artifact in client.list_artifacts(run_id=run.info.run_id):
+                # Get the artifact's local path (MLFlow suggests that if the artifact is already in the local filesystem
+                # its local path will be returned:
+                artifact_local_path = mlflow.artifacts.download_artifacts(
+                    run_id=run.info.run_id,
+                    artifact_path=artifact.path,
+                )
+                # Check if the artifact is a model (will be logged after the artifacts):
+                if artifact.is_dir and os.path.exists(
+                    os.path.join(artifact_local_path, "MLmodel")
```

Comment:
what is this `MLmodel` directory? is that given by us? fixed by mlflow?

Reply:
yes, it is ALWAYS the case.

Reply:
it's a tag to show this dir is the model

Reply:
> yes, it is ALWAYS the case.

that worth a comment